src/NUnitEngine/nunit.engine/Services/ExtensionService.cs

```diff
@@ -434,6 +447,7 @@ internal void FindExtensionsInAssembly(ExtensionAssembly assembly)
                     return;
                 }
             }
+#endif
```

Comment:
I added a unit test for this where the .NET Core 2.0 targeted tests try to load extensions from the .NET 3.5 assembly and visa-versa. Interestingly, .NET 3.5 correctly throws an `NUnitEngineException` because it thinks the .NET Core 2.0 targets the 4.0 full framework. The .NET Core 2.0 seems to successfully load the .NET 3.5 extensions. Both of these behaviors seem wrong to me.